lms/djangoapps/commerce/utils.py

```diff
@@ -48,7 +48,13 @@ def is_enabled(self, user):
         """
         Determines the availability of the Ecommerce service based on user activation and service configuration.
         """
-        return user.is_active and self.config.checkout_on_ecommerce_service
+        return user.is_active and self.is_ecommerce_checkout_enabled()
+
+    def is_ecommerce_checkout_enabled(self):
```

Comment:
Hmm, it might be a bit confusing on the calling side as far as the difference between is_enabled and is_ecommerce_checkout_enabled.  I still think we should work with is_enabled -- is it possible to add a kwarg such as "allow_anonymous" which would control the check for user.is_active?  We could default the kwarg to False so that the normal workflow would be to perform the user.is_active check.